compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstPropertyChecker.kt

```diff
@@ -40,9 +45,19 @@ object FirConstPropertyChecker : FirPropertyChecker() {
             reporter.reportOn(declaration.delegate?.source, FirErrors.CONST_VAL_WITH_DELEGATE, context)
         }
 
-        // TODO: Implement checkers for these errors (see ConstModifierChecker in FE1.0):
-        // - TYPE_CANT_BE_USED_FOR_CONST_VAL
-        // - CONST_VAL_WITHOUT_INITIALIZER
-        // - CONST_VAL_WITH_NON_CONST_INITIALIZER
+        val initializer = declaration.initializer
+        if (initializer == null) {
+            reporter.reportOn(declaration.source, FirErrors.CONST_VAL_WITHOUT_INITIALIZER, context)
+            return
+        }
+
+        if (!declaration.returnTypeRef.coneType.canBeUsedForConstVal()) {
+            reporter.reportOn(declaration.source, FirErrors.TYPE_CANT_BE_USED_FOR_CONST_VAL, declaration.returnTypeRef.coneType, context)
```

Comment:
Looks like this error should not be reported for error types (because other errors are reported in this case). Please fix this.

Reply:
Also, some usable types may be wrapped here in `ConeFlexibleType`, like `String!`